src/Compilers/CSharp/Portable/Lowering/Instrumentation/StackOverflowProbingInstrumenter.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.Shared.Collections;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+/// <summary>
+/// Implements instrumentation for <see cref="CodeAnalysis.Emit.InstrumentationKind.StackOverflowProbing"/>.
+/// </summary>
+internal sealed class StackOverflowProbingInstrumenter(
+        MethodSymbol ensureStackMethod,
+        SyntheticBoundNodeFactory factory,
+        Instrumenter previous)
+        : CompoundInstrumenter(previous)
+{
+    protected override CompoundInstrumenter WithPreviousImpl(Instrumenter previous)
+        => new StackOverflowProbingInstrumenter(
+            ensureStackMethod,
```

Comment:
>ensureStackMethod [](http://example.com/codeflow?start=12&length=17)

We do not allow implicit capturing of primary constructor parameters under Compilers.